lib/libutils/ext/mempool.c

```diff
@@ -117,47 +139,25 @@ mempool_alloc_pool(void *data, size_t size,
 
 void *mempool_alloc(struct mempool *pool, size_t size)
 {
-	size_t offset;
-	struct mempool_item *new_item;
-	struct mempool_item *last_item = NULL;
+	void *p = NULL;
 
 	get_pool(pool);
 
-	if (pool->last_offset < 0) {
-		offset = 0;
-	} else {
-		last_item = (struct mempool_item *)(pool->data +
-						    pool->last_offset);
-		offset = pool->last_offset + last_item->size;
-
-		offset = ROUNDUP(offset, MEMPOOL_ALIGN);
-		if (offset > pool->size)
-			goto error;
-	}
-
-	size = sizeof(struct mempool_item) + size;
-	size = ROUNDUP(size, MEMPOOL_ALIGN);
-	if (offset + size > pool->size)
-		goto error;
-
-	new_item = (struct mempool_item *)(pool->data + offset);
-	new_item->size = size;
-	new_item->prev_item_offset = pool->last_offset;
-	if (last_item)
-		last_item->next_item_offset = offset;
-	new_item->next_item_offset = -1;
-	pool->last_offset = offset;
+	p = raw_malloc(0, 0, size, pool->mctx);
+	if (p) {
 #ifdef CFG_MEMPOOL_REPORT_LAST_OFFSET
-	if (pool->last_offset > pool->max_last_offset) {
-		pool->max_last_offset = pool->last_offset;
-		DMSG("Max memory usage increased to %zu",
-		     (size_t)pool->max_last_offset);
-	}
+		struct malloc_stats stats = { };
```

Comment:
This code block should also be in CFG_WITH_STATS along with CFG_MEMPOOL_REPORT_LAST_OFFSET

Reply:
I'll enable CFG_WITH_STATS from `mk/config.mk` when `CFG_MEMPOOL_REPORT_LAST_OFFSET=y`.